.github/workflows/pr-build.yml

```diff
@@ -80,6 +80,11 @@ jobs:
           gpg_private_key: ${{ secrets.GPG_PRIVATE_KEY }}
           gpg_password: ${{ secrets.GPG_PASSPHRASE }}
 
+      - name: Dry Run Validator
+        if: ${{ matrix.os != 'windows-latest' }} # Skip dry run validator on windows as it is not possible to build opentelemetry-java on windows
+        run: |
+          ./gradlew testing:validator:build
+
```

Comment:
This should be in it's own job so it can run independently. Ideally this would only be ran when validator files change but I'm not sure how much time that would save us.

Reply:
Since validator and the other code sharing same gradle file, this test added here aimed to avoid a case that the validator not change and the code repo changed to broke the validator.

Reply:
Can we use the gradle-build-action instead? That way we can utilize the caching by default.